LEAF_Request_Portal/templates/reports/LEAF_mass_action.tpl

```diff
@@ -89,11 +95,818 @@
+        },
+        async: false
+    });
+
+    return steps;
+}
+
+/**
+ * Purpose: need to get the form types and put them in the form_container as a drop down
+ */
+
+function getForms() {
+    let forms;
+
+    $.ajax({
+        type: "GET",
+        url: "./api/workflow/categoriesUnabridged",
+        cache: false,
+        success: function (res) {
+            forms = res;
+        },
+        error: function (err) {
+            console.log(err);
+        },
+        async: false
+    });
+
+    return forms;
+}
+
+/**
+ * Purpose do reminder search (used by click or change of lastAction text)
+ */
+function reminderDaysSearch() {
+    let daysSince = document.getElementById("lastAction").valueOf();
+    doSearch();
+}
+
+/**
+ * Sets up and builds the search query, passing it along to listRequests
+ */
+function doSearch() {
+    let getCancelled = false;
+    let getSubmitted = true;
+    let getReminder = 0;
+    let getAction = "";
+
+    $("input#selectAllRequests").prop("checked", false);
+    setProgress("");
+    // Get Dropdown values
+    actionValue = $("select#action").val();
+    switch (actionValue) {
+        case "email":
+            getReminder = Number(document.getElementById("lastAction").value);
+            break;
+        case "submit":
+            getSubmitted = false;
+            break;
+        case "restore":
+            getCancelled = true;
+            break;
+        case "take_action":
+            getAction = document.getElementById("action_select").value;
+            break;
+    }
+
+    let queryObj = buildQuery(getCancelled, getSubmitted, getReminder, getAction);
+    searchID = Math.floor(Math.random() * 1000000000);
+    listRequests(queryObj, searchID, getReminder);
+}
+
+/**
+ * Builds query object to pass to form/query
+ *
+ * @param {boolean}                [getCancelled]                                 filter by cancelled
+* @param {boolean}                [getSubmitted]                                 filter by submitted
+* @param {int}                                                [getReminder]                                                value of email reminder selection
+*
+* @return {Object} query object to pass to form/query.
+*/
+function buildQuery(getCancelled, getSubmitted, getReminder, getAction) {
+    let requestQuery = {
+        terms: [],
+        joins: ["service", "recordsDependencies", "categoryName", "status"],
+        sort: {},
+    };
+
+    if (getCancelled) {
+        requestQuery.terms.push({
+            id: "stepID",
+            operator: "=",
+            match: "deleted",
+        });
+    } else if (!getAction) {
+        requestQuery.terms.push({
+            id: "stepID",
+            operator: "!=",
+            match: "deleted",
+        });
+    }
+
+    if (!getSubmitted) {
+        requestQuery.terms.push({
+            id: "stepID",
+            operator: "!=",
+            match: "submitted",
+        });
+    }
+
+    if (getReminder) {
```

Comment:
getReminder is an int, not a boolean.

Reply:
I didn't add this but updated it to > 0. Also if a variable is a number it will return true as long as it's not 0, 0 returns false.